### ISSUE TITLE
no local_dss_weights method for LevelGrid

### ISSUE THREAD

**Issue by juliasloan25** [CliMA/ClimaCore.jl#1565]
## Describe the bug
I'm getting a bug in `Spaces.weighted_dss!(x)`, where `x::Fields.Field` when updating to ClimaCore v0.11.0. 

I think this is from a similar cause as #1564 , where a space created using `Spaces.level` has `grid(space) isa LevelGrid`, even when the space is ExtrudedSpace. Then the space (and grid) are passed into a function that dispatches on `AbstractSpectralElementSpace` and uses functionality the `LevelGrid` doesn't have.

I can't print the space or grid because `show` for these types goes to [this line](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Spaces/spectralelement.jl#L32), and gives `ERROR: LoadError: type LevelGrid has no field quadrature_style`.
 
## To Reproduce
```julia
import ClimaCore: Spaces, Domains, Meshes, Geometry, Topologies, Spaces, Fields
using ClimaCore.Utilities: PlusHalf
using IntervalSets

function hvspace_2D(xlim = (-π, π), zlim = (0, 4π), helem = 20, velem = 20, npoly = 1)
    FT = Float64
    vertdomain = Domains.IntervalDomain(
        Geometry.ZPoint{FT}(zlim[1]),
        Geometry.ZPoint{FT}(zlim[2]);
        boundary_tags = (:bottom, :top),
    )
    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = velem)
    vert_center_space = Spaces.CenterFiniteDifferenceSpace(vertmesh)

    horzdomain = Domains.IntervalDomain(Geometry.XPoint{FT}(xlim[1]) .. Geometry.XPoint{FT}(xlim[2]), periodic = true)
    horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
    horztopology = Topologies.IntervalTopology(horzmesh)
    quad = Spaces.Quadratures.GLL{npoly + 1}()
    horzspace = Spaces.SpectralElementSpace1D(horztopology, quad)

    hv_center_space = Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
    return (hv_center_space, hv_face_space)
end

## construct domain spaces
hv_center_space, hv_face_space = hvspace_2D((-500, 500), (0, 1000), 20, 20, 4) # [m]
domain = (hv_center_space = hv_center_space, hv_face_space = hv_face_space)

# TODO this gives error in this line when run in REPL
# https://github.com/CliMA/ClimaCore.jl/blob/main/src/Spaces/spectralelement.jl#L32
atm_boundary = Spaces.level(domain.hv_face_space, PlusHalf(0))

# TODO this also fails (in REPL or not) here:
# https://github.com/CliMA/ClimaCore.jl/blob/main/src/Spaces/spectralelement.jl#L10
Spaces.weighted_dss!(Fields.ones(atm_boundary))
```
<details><summary>Project</summary>

```
(sea_breeze) pkg> st
Status `~/clima/ClimaCoupler.jl/experiments/ClimaCore/sea_breeze/Project.toml`
⌃ [d414da3d] ClimaCore v0.11.0
  [4ade58fe] ClimaCoupler v0.1.0 `../../..`
  [459566f4] DiffEqCallbacks v2.34.0
  [8197267c] IntervalSets v0.7.8
⌃ [1dea7af3] OrdinaryDiffEq v6.58.2
  [91a5bcdd] Plots v1.39.0
  [08abe8d2] PrettyTables v2.3.1
⌅ [0bca4576] SciMLBase v1.98.1
  [90137ffa] StaticArrays v1.7.0
  [5d786b92] TerminalLoggers v0.1.7
  [b60c26fb] Thermodynamics v0.11.2
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [10745b16] Statistics v1.9.0
```
</details>

## System details

Any relevant system information:
- Julia v1.9.4
- macOS

## Related issues / PRs
related to #1564

**Comment by simonbyrne** [comment]
@juliasloan25 do you want to have a stab at fixing this?

**Comment by juliasloan25** [comment]
closed by #1569
